libcontainer/dmz/cloned_binary_linux.go

```diff
@@ -104,6 +104,9 @@ func mktemp(dir string) (*os.File, SealFunc, error) {
 	if err != nil {
 		return nil, nil, err
 	}
+	if err := file.Chmod(0o511); err != nil {
+		return nil, nil, fmt.Errorf("chmod classic tmpfile: %w", err)
+	}
```

Comment:
Isn't this already done in `sealFile`?

Reply:
Yes, but because of `isExecutable` check, we will get an error because we can't find any executable file after we traversed all the `tmpDirs`:
https://github.com/opencontainers/runc/blob/e2647bd759734d71633f95eb7b51c5806e09b537/libcontainer/dmz/cloned_binary_linux.go#L165-L169

Reply:
Hmm, okay so we should move the code from `sealFile` then rather than duplicating it. Also we should do this after we do `os.Remove()` so if there is an error we don't leak the temporary file.

Reply:
It would be nice to explain the issue in the commit message btw :smile_cat: